src/renderer/viz/expressions/ramp.js

```diff
@@ -325,6 +342,35 @@ export default class Ramp extends BaseExpression {
         gl.uniform1f(this._getBinding(program).keyMinLoc, (this.minKey));
         gl.uniform1f(this._getBinding(program).keyWidthLoc, (this.maxKey) - (this.minKey));
         drawMetadata.freeTexUnit++;
+
+        if (this.input.type === 'category' && this.input.isA(Property)) {
+            gl.activeTexture(gl.TEXTURE0 + drawMetadata.freeTexUnit);
+            if (this._translatedIds !== this._metadata.properties[this.input.name].categories.length) {
+                this._translatedIds = this._metadata.properties[this.input.name].categories.length;
+                this._translateTexture = gl.createTexture();
+                const translatorPixels = new Float32Array(256 * 256);
```

Comment:
Instead of 256, we have `COLOR_ARRAY_LENGTH`.

Reply:
Ups, this constant should be extracted and named, although it is not related to that one. It is just the maximum number of categories per property.